src/rule_definition_tools.jl

```diff
@@ -80,16 +80,16 @@ macro scalar_rule(call, maybe_setup, partials...)
         end
     end
     if all(Meta.isexpr(partial, :tuple) for partial in partials)
-        forward_rules = Any[rule_from_partials(partial.args...) for partial in partials]
+        forward_rules = Any[rule_from_partials(:(promote($(inputs...))[1]), partial.args...) for partial in partials]
```

Comment:
I think break this on to two lines
so can give a little explination.
Also `first` has a (negligable) performance advantage over `xs[1]` for tuples.
(because it can solve inbounds at the type level)

```suggestion
        input_rep = :(first(promote($(inputs...))))  # stand-in with the right type for an input
        forward_rules = Any[rule_from_partials(input_rep, partial.args...) for partial in partials]
```

Reply:
Note that this will not work for the 0-arg case, but I'm wondering whether we should just leave it for now